pvanalytics/quality/irradiance.py

```diff
@@ -471,3 +471,185 @@ def daily_insolation_limits(irrad, clearsky, daily_min=0.4, daily_max=1.25):
+    # component sum series.
+    # Find the locations where the sun is below the sza limit.
+    mask = (zenith_limit <= solar_zenith)
+    if isinstance(fill_night_value, float) | isinstance(fill_night_value, int):
+        # Replace the nighttime values with a fill value--this can be np.nan,
+        # which is a float
+        series[mask] = fill_night_value
+    elif fill_night_value == 'equation':
+        # Use the nighttime equation GHI = 0 + DHI, which translates as:
+        # GHI_Calc (at night) = DHI_measured
+        # DHI_Calc (at night) = GHI_measured
+        # DNI_Calc (at night) = 0
+        if component == 'GHI':
+            series[mask] = component_sum_df['dhi'][mask]
+        elif component == 'DHI':
+            series[mask] = component_sum_df['ghi'][mask]
+        else:
+            series[mask] = 0
+    elif fill_night_value is None:
+        pass
+    else:
+        raise ValueError("The fill_night_value variable must be None,"
+                         " float or int, or 'equation'. Please change "
+                         "the variable fill_night_value value.")
+    return series
+
+
+def _complete_irradiance(solar_zenith,
+                         ghi=None,
+                         dhi=None,
+                         dni=None,
+                         dni_clear=None):
+    """
+    TODO: This method exists in the pvlib-python library. Once a new PVLib
+    release or pre-release is cut, this private function can be deleted and
+    the associated PVLib function can be directly leveraged.
+
+    Use the component sum equations to calculate the missing series, using
+    the other available time series. One of the three parameters (ghi, dhi,
+    dni) is passed as None, and the other associated series passed are used to
+    calculate the missing series value.
+    The "component sum" or "closure" equation relates the three
+    primary irradiance components as follows:
+    .. math::
+       GHI = DHI + DNI * cos(theta_z)
+    Parameters
+    ----------
+    solar_zenith : Series
+        Zenith angles in decimal degrees, with datetime index.
+        Angles must be >=0 and <=180. Must have the same datetime index
+        as ghi, dhi, and dni series, when available.
+    ghi : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as dni, dhi, and zenith series, when available.
+    dhi : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dni, and zenith series, when available.
+    dni : Series, optional
+        Pandas series of dni data, with datetime index. Must have the same
+        datetime index as ghi, dhi, and zenith series, when available.
+    dni_clear : Series, optional
+        Pandas series of clearsky dni data. Must have the same datetime index
+        as ghi, dhi, dni, and zenith series, when available. See
+        :py:func:`dni` for details.
```

Comment:
```suggestion
        as `ghi`, `dhi`, `dni`, and `solar_zenith` series, when available. See pvlib-python's 
        [dni](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.dni.html#pvlib.irradiance.dni) for details.
```

Reply:
We have pvlib's docs linked via intersphinx, so I think ``` :py:func:`pvlib.irradiance.dni` ``` should work.  Although this docstring doesn't actually get rendered as HTML anywhere, so maybe @cwhanse's suggestion is better since this docstring will only ever be viewed as plaintext.